networking.go

```diff
@@ -0,0 +1,149 @@
+package veneur
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/url"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// ListeningAddr implements the net.Addr interface and gets
+// deserialized from the YAML config file by interpreting it as a URL,
+// where the Scheme corresponds to the "net" argument to net.Listen,
+// and the host&port or path are the "laddr" arg.
+//
+// Valid address examples are:
+//   - udp6://127.0.0.1:8000
+//   - unix:///tmp/foo.sock
+//   - tcp://127.0.0.1:9002
+type ListeningAddr struct {
```

Comment:
What's the advantage of using a struct here instead of `net.Addr` directly (or a new named type for `net.Addr`)?